src/TransitFunctionAppTest/ModelsTests/PurchaseTicketRequestTest.cs

```diff
@@ -17,19 +19,48 @@ public void SetUp()
         public void PurchaseTicketTest()
         {
             // sample payload to test 
-            var payload = @"{'DeviceId':'rjTest','DeviceType':'Kiosk','MessageType':'Purchase','TransactionId':'c3d8b13c - 3e8f - 4e57 - a7d5 - ae8adaa8c2b2','CreateTime':'2018 - 10 - 11T20: 49:58.0667867Z','OriginLocation':null,'DestinationLocation' :null,'DepartureTime':'0001 - 01 - 01T00: 00:00','Price':'54','MethodName':'SetTelemetryInterval'}";
+            PurchaseTicketRequest sampleRequest = new PurchaseTicketRequest()
+            {
+                DeviceId = "rjTest",
+                DeviceType = DeviceType.TicketKiosk,
+                TransactionId = "c3d8b13c - 3e8f - 4e57 - a7d5 - ae8adaa8c2b2",
+                CreateTime = System.DateTime.UtcNow,
+                Price = 54,
+                MethodName = "ReceivePurchaseTicketResponse"
+            };
+            string sampleRequestString = JsonConvert.SerializeObject(sampleRequest);
+
+            // expected response 
+            PurchaseTicketPayload expectedResponse = new PurchaseTicketPayload()
+            {
+                DeviceId = sampleRequest.DeviceId,
+                TransactionId = sampleRequest.TransactionId,
+                DeviceType = sampleRequest.DeviceType,
+                MessageType = MessageType.cmdPurchaseTicket,
+                IsApproved = true
+            };
+            string expectedResponseString = JsonConvert.SerializeObject(expectedResponse);
+
+
             FakeInvokeDeviceMethod serviceClient = new FakeInvokeDeviceMethod();
-            PurchaseTicketAction action = new PurchaseTicketAction(serviceClient, payload, logger);
+            PurchaseTicketAction action = new PurchaseTicketAction(serviceClient, 0, sampleRequestString, logger);
```

Comment:
@Sarah-Aly I've made some updates to the unit test. This was in part to account for the "jitter" in the failure rates as the test would occasionally fail but work on a re-run. I finally realized this was due to the random "deny purchase" chance. So I added a jitter value to the action constructor so we could control that during testing as well as make the real value based on configuration settings and not hard-coded (the later part still needs to be implemented). 

I also updated the unit test to avoid comparing the entire string. I feel this helps call out specifically which values were not right as well as help ensures the test remains constant between changes to the various message payloads. 

I'd appreciate your thoughts.